File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.cpp
class Solution {
public:
    int maximumCount(vector<int>& nums) {
        int n = nums.size();
        int l = 0, r = n - 1;
        // Find the first index with a positive number
        while (l <= r) {
            int m = (l + r) / 2;
            if (nums[m] <= 0) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }
        int posCount = n - l;
        
        // Find the last index with a negative number
        l = 0;
        r = n - 1;
        while (l <= r) {
            int m = (l + r) / 2;
            if (nums[m] < 0) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }
        int negCount = r + 1;
        
        return max(posCount, negCount);
    }
};

File: 2614-maximum-count-of-positive-integer-and-negative-integer/maximum-count-of-positive-integer-and-negative-integer.py
from typing import List

class Solution:
    def maximumCount(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        # Find the first index with a positive number
        while l <= r:
            m = (l + r) // 2
            if nums[m] <= 0:
                l = m + 1
            else:
                r = m - 1
        posCount = len(nums) - l
        
        l, r = 0, len(nums) - 1
        # Find the last index with a negative number
        while l <= r:
            m = (l + r) // 2
            if nums[m] < 0:
                l = m + 1
            else:
                r = m - 1
        negCount = r + 1
        
        return max(posCount, negCount)
